libs/ui-lib/lib/common/components/ui/formik/CodeField.tsx

```diff
@@ -26,7 +27,30 @@ const CodeField = ({
   const [monacoSubscription, setMonacoSubscription] = React.useState<monaco.IDisposable>();
   const [monacoEditor, setMonacoEditor] = React.useState<monaco.editor.IStandaloneCodeEditor>();
   const errorMessage = useFieldErrorMsg({ name, validate });
-
+  const codeEditorRef = React.useRef(null);
+  const [isValidValue, setIsValidValue] = React.useState<boolean>(true);
+  const [typingTimeout, setTypingTimeout] = React.useState<NodeJS.Timeout | null>(null);
+  const onCodeChange = (value: string) => {
+    if (typingTimeout) {
+      clearTimeout(typingTimeout);
+      setIsValidValue(true);
+    }
+    const typingTimeout_new = setTimeout(() => {
+      let isValidVal = true;
+      if (codeEditorRef && codeEditorRef.current) {
+        if ((codeEditorRef.current as CodeEditor).state.filename) {
+          isValidVal = validateFileName((codeEditorRef.current as CodeEditor).state.filename);
+          setIsValidValue(isValidVal);
+        }
+      }
+      if (!isValidVal) {
+        setValue('', true);
+      } else {
+        setValue(value, true);
+      }
+    }, 500);
+    setTypingTimeout(typingTimeout_new);
+  };
```

Comment:
Scenario: I am a user trying to upload some valid yaml content and I am very confused.

https://github.com/openshift-assisted/assisted-installer-ui/assets/87187179/f7717adb-4a6e-4e3e-8787-d8948e625fc8

I don't think enforcing a file extention when we only care about the content is a good approach. Also, this should be done at form level, not on CodeField component level.